test/ReactWrapper-spec.jsx

```diff
@@ -968,6 +968,23 @@ describeWithDOM('mount', () => {
       expect(setInvalidProps).to.throw(TypeError, similarException.message);
     });
 
+    it('should call the callback when setProps has completed', () => {
+      class Foo extends React.Component {
+        render() {
+          return (
+            <div className={this.props.id}>
+              {this.props.id}
+            </div>
+          );
+        }
+      }
+      const wrapper = mount(<Foo id="foo" />);
+      expect(wrapper.find('.foo').length).to.equal(1);
+      wrapper.setProps({ id: 'bar', foo: 'bla' }, () => {
+        expect(wrapper.find('.bar').length).to.equal(1);
+      });
```

Comment:
is there a way we could add a sync assertion that the length is NOT equal to 1, to prove that the callback is doing its job?